mw_strip_auth.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/Sirupsen/logrus"
+)
+
+type StripAuth struct {
+	BaseMiddleware
+}
+
+func (sa *StripAuth) Name() string {
+	return "StripAuth"
+}
+
+func (sa *StripAuth) EnabledForSpec() bool {
+	return sa.Spec.StripAuthData
+}
+
+func (sa *StripAuth) ProcessRequest(w http.ResponseWriter, r *http.Request, _ interface{}) (error, int) {
+
+	log.WithFields(logrus.Fields{
+		"prefix": sa.Name(),
+	}).Debugf("sa.Spec.Auth: %+v\n", sa.Spec.Auth)
+
+	if sa.Spec.Auth.UseParam {
+		if err := sa.stripFromParams(r); err != nil {
+
+			// TODO - code review, confirm how to handle error:
+			// return nil, 200 // but log msg to allow request to proceed
+			// return err, 200
+			return err, 500
+		}
+
+		return nil, 200
+	}
+
+	sa.stripFromHeaders(r)
+
+	return nil, 200
+}
+
+func (sa *StripAuth) stripFromParams(r *http.Request) error {
+
+	config := sa.Spec.Auth
+
+	reqUrlPtr, err := url.Parse(r.URL.String())
+	if err != nil {
+		return err
```

Comment:
I think it should be totally safe ignore both of this errors in this function. In the first case, you are parsing current URL, which was already validated. In the second case, almost the same URL object, just changed query string a bit.